File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
answer = 0

def dfs(idx, now_cal, nums, key):
    global answer
    
    if idx < len(nums):
        dfs(idx + 1, now_cal + nums[idx], nums, key)
        dfs(idx + 1, now_cal - nums[idx], nums, key)
    else:
        if now_cal == key:
            answer += 1
    
def solution(numbers, target):
    global answer
    dfs(0, 0, numbers, target)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int ret=0;

void dfs (int cur, int val, int goal, vector<int>& nums) {
    if (cur == nums.size()) {
        if (val == goal) {
            ret++;   
        }
        return ;
    }
    
    dfs(cur+1, val + nums[cur]*(-1), goal, nums);
    dfs(cur+1, val + nums[cur], goal, nums);
    
}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    dfs(0, 0, target, numbers);
    
    return ret;
}